From 0d11be6f98e80b04e0d16c06d16889ee8cf320de Mon Sep 17 00:00:00 2001
From: Shinichi Yamashita <yamashitasni@oss.nttdata.co.jp>
Date: Thu, 18 Jun 2015 19:59:00 -0700
Subject: [PATCH 0479/1164] HIVE-11135 : Fix the Beeline set and save command
 in order to avoid the NullPointerException
 (Shinichi Yamashita via Xuefu Zhang)

Change-Id: If1a88446f91f8f2db47990f5fb04eb529d984c3a
Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>
---
 .../java/org/apache/hive/beeline/BeeLineOpts.java  |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java b/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
index 894f74f..63a0a56 100644
--- a/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
+++ b/beeline/src/java/org/apache/hive/beeline/BeeLineOpts.java
@@ -200,9 +200,9 @@ public Properties toProperties()
 
     String[] names = propertyNames();
     for (int i = 0; names != null && i < names.length; i++) {
+      Object o = beeLine.getReflector().invoke(this, "get" + names[i], new Object[0]);
       props.setProperty(PROPERTY_PREFIX + names[i],
-          beeLine.getReflector().invoke(this, "get" + names[i], new Object[0])
-              .toString());
+          o == null ? "" : o.toString());
     }
     beeLine.debug("properties: " + props.toString());
     return props;
-- 
1.7.9.5

